### ISSUE TITLE
Add max_split option to `pl.col().str.split(by: str, inclusive: bool = False, max_split: int)`

### ISSUE THREAD

**Issue by ghuls** [pola-rs/polars#2746]
#### Describe your feature request

Add max_split option to `pl.col().str.split(by: str, inclusive: bool = False, max_split: int)`

If a string can' be split in enough pieces, add None for those parts.

Also a way to add those splitted strings as separated columns would be great. As the number of splits is known, adding extra columns should be easy and no pl.List needs to be added for the column.

Maybe something like this:
```python
pl.col("a").str.split(by= "_", max_split=3, with_columns=["a1", "a2", "a3"])
```

**Comment by moritzwilksch** [comment]
Not sure whether this is what you're looking for, but this effect can be achieved like this:

```python
>>> df = pl.DataFrame(
...     {
...         "s": ["a b c", "d e f", "g h i"],
...     }
... )
>>> 
>>> df.with_columns(pl.col("s").str.split(" ").alias("split")).explode("split").groupby(
...     "s"
... ).agg([pl.col("split").take(idx).alias(f"split_at_{idx}") for idx in range(3)])
shape: (3, 4)
┌───────┬────────────┬────────────┬────────────┐
│ s     ┆ split_at_0 ┆ split_at_1 ┆ split_at_2 │
│ ---   ┆ ---        ┆ ---        ┆ ---        │
│ str   ┆ str        ┆ str        ┆ str        │
╞═══════╪════════════╪════════════╪════════════╡
│ d e f ┆ d          ┆ e          ┆ f          │
├╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌┤
│ g h i ┆ g          ┆ h          ┆ i          │
├╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌┤
│ a b c ┆ a          ┆ b          ┆ c          │
└───────┴────────────┴────────────┴────────────┘
```

**Comment by ghuls** [comment]
I know it can be done this way, but if polars knows that the split will always result in the same number of columns it can avoid doing the explode and groupby operations.

**Comment by moritzwilksch** [comment]
Ahh I see!

**Comment by ritchie46** [comment]
It is not easy to assign these columns to the `DataFrame`. An expression is `Fn(Expr) -> Expr`. This means we cannot deal multiple yielded expressions.

What is an option is adding a computational node to the `DataFrame/LazyFrame`. Something like `explode` but then horizontally. This could explode a list like

```
[[1, 2, 3],
[4, 5, 6]]
```

To a `DataFrame`

```
column_1, column_2, column_3
1, 2, 3
4, 5, 6
```

**Comment by ritchie46** [comment]
Closed by #2872
